chiselFrontend/src/main/scala/chisel3/Printable.scala

```diff
@@ -157,22 +161,40 @@ object FirrtlFormat {
   }
 }
 /** Format bits as Decimal */
-case class Decimal(bits: Bits) extends FirrtlFormat('d')
+case class Decimal(bits: Bits) extends FirrtlFormat('d') {
+  override def length: Int = bits.widthOption match {
+    case Some(w) => Math.ceil(Math.log10(Math.pow(2, w) + 1)).toInt
+    case None => 0
+  }
+}
```

Comment:
Would you mind factoring out these changes from this PR? Or: are these critical to the functioning of this PR?

Reply:
No, they aren't necessary. They are only useful for one of my aspect libraries. I can factor it out.